lib/supavisor.ex

```diff
@@ -124,11 +125,22 @@ defmodule Supavisor do
     }
   end
 
-  @spec get_local_pool(id) :: pid | nil
+  @spec get_local_pool(id) :: map | pid | nil
```

Comment:
Would it be more consistent to always return a map (which may also be empty?).

Reply:
I guess though you want to distinguish if it is a read-write cluster or a "shared" one. So perhaps the current return type is fine.

Reply:
Yep, that was my intention. The problem is I'm trying to implement such a massive update without rewriting all the existing code which can make some areas appear odd and unclear

Reply:
If it helps, something that I like to do in such cases is to do code breaking changes first and rewrite the usage without adding new features. Only then I would add the new features.

Of course, it all depends on you finding out that the API is going to change early in the process, which is not always the case. In this case, it is probably not worth doing it, since the work is well underway. :)

Reply:
That's a very smart approach 🤔🤔🤔